include/openrave/openravejson.h

```diff
@@ -409,6 +443,40 @@ inline void LoadJsonValue(const rapidjson::Value& v, std::pair<T, U>& t) {
     }
 }
 
+inline void LoadJsonValue(const rapidjson::Value& v, OpenRAVE::TriMesh& t)
+{
+    if (!v.IsObject()) {
+        throw OPENRAVE_EXCEPTION_FORMAT0("Cannot load value of non-object.", OpenRAVE::ORE_InvalidArguments);
+    }
+
+    if (!v.HasMember("vertices") || !v["vertices"].IsArray() || v["vertices"].Size() % 3 != 0) {
+        throw OPENRAVE_EXCEPTION_FORMAT0("failed to deserialize json, value cannot be decoded as a TriMesh, \"vertices\" malformatted", OpenRAVE::ORE_InvalidArguments);
+    }
+
+    t.vertices.clear();
+    t.vertices.reserve(v["vertices"].Size() / 3);
+
+    for (rapidjson::Value::ConstValueIterator it = v["vertices"].Begin(); it != v["vertices"].End(); ) {
+        OpenRAVE::Vector vertex;
+        LoadJsonValue(*(it++), vertex.x);
+        LoadJsonValue(*(it++), vertex.y);
+        LoadJsonValue(*(it++), vertex.z);
+        t.vertices.push_back(vertex);
+    }
+
+    if (!v.HasMember("indices") || !v["indices"].IsArray()) {
+        throw OPENRAVE_EXCEPTION_FORMAT0("failed to deserialize json, value cannot be decoded as a TriMesh, \"indices\" malformatted", OpenRAVE::ORE_InvalidArguments);
+    }
+
+    t.indices.resize(v["indices"].Size());
```

Comment:
what is your reason to change from `LoadJsonValue(v["indices"], t.indices);`?

Reply:
@yoshikikanemoto it is because `t.indices` is a `std::vector<int32_t>` and  `LoadJsonValue` for `std::vector<T>` is defined after `LoadJsonValue` for `OpenRAVE::TriMesh& t)`. Otherwise we will get an error at compile time.

- https://stackoverflow.com/questions/7767626/two-phase-lookup-explanation-needed#comment9456869_7767726

Reply:
@yoshikikanemoto Oh my previous comment is not correct. For this one, I didn't need change this line because we have forward declaration for `LoadJsonValue` for `std::vector<T>`. I needed this change for https://github.com/rdiankov/openrave/pull/1213#discussion_r1148432877 but that one is also avoidable by adding some forward declaration.

Let me update this PR to minimize the changes.

Reply:
@yoshikikanemoto updated. Please check again when you have time.